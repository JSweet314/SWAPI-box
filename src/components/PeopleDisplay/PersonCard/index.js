import React, { Component } from 'react';
import './style.css';
import PropTypes from 'prop-types';

const PersonCard = ({card, favorites}) => {
  const selected = favorites.some(favorite => favorite.name === card.name) ?
    'selected' : '';

  const commasAdded = (number) => {
    return number.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
  }

  return (
    <article className="info-card">
      <div>
        <div className="info-card__top-line">
          <h3>{card.name}</h3>
          <button className={selected}></button>
        </div>
        <p>Homeworld: {card.homeworld}</p>
        <p>Species: {card.species}</p>
        <p>Home Population: {commasAdded(card.population)}</p>
      </div>
    </article>
  );
};
  
PersonCard.propTypes = {
  card: PropTypes.shape({
    name: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    homeworld: PropTypes.string.isRequired,
    population: PropTypes.string.isRequired
  }).isRequired,
  removeFavorite: PropTypes.func.isRequired,
  selectFavorite: PropTypes.func.isRequired,
  changeFavCount: PropTypes.func.isRequired,
  currentCategory: PropTypes.string.isRequired
};

export default PersonCard;