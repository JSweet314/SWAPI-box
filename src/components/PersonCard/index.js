import React from 'react';
import PropTypes from 'prop-types';
import {addCommas} from '../../helpers/helper';
import './style.css';

const PersonCard = ({card, favorites, handleFavoriteClick}) => {
  const selected = favorites.some(favorite => favorite.name === card.name) ?
    'selected' : '';

  return (
    <article className="info-card">
      <div>
        <div className="info-card__top-line">
          <h3>{card.name}</h3>
          <button 
            className={selected}
            onClick={() => handleFavoriteClick(card, 'people')}>
          </button>
        </div>
        <p>Homeworld: {card.homeworld}</p>
        <p>Species: {card.species}</p>
        <p>Home Population: {addCommas(card.population)}</p>
      </div>
    </article>
  );
};
  
PersonCard.propTypes = {
  card: PropTypes.shape({
    name: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    homeworld: PropTypes.string.isRequired,
    population: PropTypes.string.isRequired
  }).isRequired,
  favorites: PropTypes.array.isRequired,
  handleFavoriteClick: PropTypes.func.isRequired
};

export default PersonCard;