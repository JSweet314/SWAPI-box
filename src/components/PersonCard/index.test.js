import React from 'react';
import {shallow} from 'enzyme';
import PersonCard from './index';

describe('PersonCard', () => {
  /* eslint-disable no-undef */
  const mockHandleOnClick = jest.fn();
  /* eslint-enable no-undef */
  const card = {
    homeworld: 'tatooine',
    species: 'human',
    name: 'Obi-Wan',
    population: '1000'
  };

  it('should match a snapshot', () => {
    const wrapper = shallow(
      <PersonCard
        handleOnClick={mockHandleOnClick}
        favorites={[]}
        card={card} />,
      { disableLifecycleMethods: true }
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should show as favorited if name matches obj in favorites array', () => {
    const wrapper = shallow(
      <PersonCard 
        handleOnClick={mockHandleOnClick}
        favorites={[{name: 'Obi-Wan'}]}
        card={card} />,
      { disableLifecycleMethods: true }
    );

    expect(wrapper.find('.selected').length).toEqual(1);
  });
});