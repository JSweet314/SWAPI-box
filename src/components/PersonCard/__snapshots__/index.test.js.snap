// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonCard should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PersonCard
    card={
        Object {
            "homeworld": "tatooine",
            "name": "Obi-Wan",
            "population": "1000",
            "species": "human",
          }
    }
    favorites={Array []}
    handleFavoriteClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div
                className="info-card__top-line"
        >
                <h3>
                        Obi-Wan
                </h3>
                <button
                        className=""
                        onClick={[Function]}
                />
        </div>
        <p>
                Homeworld: 
                tatooine
        </p>
        <p>
                Species: 
                human
        </p>
        <p>
                Home Population: 
                1,000
        </p>
</div>,
      "className": "info-card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="info-card__top-line"
>
            <h3>
                        Obi-Wan
            </h3>
            <button
                        className=""
                        onClick={[Function]}
            />
</div>,
          <p>
            Homeworld: 
            tatooine
</p>,
          <p>
            Species: 
            human
</p>,
          <p>
            Home Population: 
            1,000
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Obi-Wan
</h3>,
              <button
                className=""
                onClick={[Function]}
/>,
            ],
            "className": "info-card__top-line",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Obi-Wan",
              },
              "ref": null,
              "rendered": "Obi-Wan",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Homeworld: ",
              "tatooine",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Homeworld: ",
            "tatooine",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Species: ",
              "human",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Species: ",
            "human",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Home Population: ",
              "1,000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Home Population: ",
            "1,000",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
                    className="info-card__top-line"
          >
                    <h3>
                              Obi-Wan
                    </h3>
                    <button
                              className=""
                              onClick={[Function]}
                    />
          </div>
          <p>
                    Homeworld: 
                    tatooine
          </p>
          <p>
                    Species: 
                    human
          </p>
          <p>
                    Home Population: 
                    1,000
          </p>
</div>,
        "className": "info-card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="info-card__top-line"
>
              <h3>
                            Obi-Wan
              </h3>
              <button
                            className=""
                            onClick={[Function]}
              />
</div>,
            <p>
              Homeworld: 
              tatooine
</p>,
            <p>
              Species: 
              human
</p>,
            <p>
              Home Population: 
              1,000
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Obi-Wan
</h3>,
                <button
                  className=""
                  onClick={[Function]}
/>,
              ],
              "className": "info-card__top-line",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Obi-Wan",
                },
                "ref": null,
                "rendered": "Obi-Wan",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld: ",
                "tatooine",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Homeworld: ",
              "tatooine",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Species: ",
                "human",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Species: ",
              "human",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Home Population: ",
                "1,000",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Home Population: ",
              "1,000",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
